avm/ptn/app/container-job-toolkit/main.bicep

```diff
@@ -0,0 +1,341 @@
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Use an existing managed identity to import the container image and run the job. If not provided, a new managed identity will be created.')
+@metadata({
+  example: '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myManagedIdentity'
+})
+param managedIdentityResourceId string?
+
+@description('Optional. The name of the managed identity to create. If not provided, a name will be generated automatically as `jobsUserIdentity-$\\{name\\}`.')
+@minLength(3)
+@maxLength(128)
+param managedIdentityName string?
+
+@description('Optional. The Log Analytics Resource ID for the Container Apps Environment to use for the job. If not provided, a new Log Analytics workspace will be created.')
+@metadata({
+  example: '/subscriptions/<00000000-0000-0000-0000-000000000000>/resourceGroups/<rg-name>/providers/Microsoft.OperationalInsights/workspaces/<workspace-name>'
+})
+param logAnalyticsWorkspaceResourceId string?
+
+@description('Optional. The connection string for the Application Insights instance that will be added to Key Vault as `applicationinsights-connection-string` and can be used by the Job.')
+@metadata({
+  example: 'InstrumentationKey=<00000000-0000-0000-0000-000000000000>;IngestionEndpoint=https://germanywestcentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://germanywestcentral.livediagnostics.monitor.azure.com/;ApplicationId=<00000000-0000-0000-0000-000000000000>'
+})
+param appInsightsConnectionString string?
+
+@description('Optional. The name of the Key Vault that will be created to store the Application Insights connection string and be used for your secrets.')
+@metadata({ example: '''kv${uniqueString(name, location, resourceGroup().name)})''' })
+@minLength(3)
+@maxLength(24)
+param keyVaultName string = 'kv${uniqueString(name, location, resourceGroup().name)}'
+
+@description('Optional. The permissions that will be assigned to the Key Vault. The managed Identity will be assigned the permissions to get and list secrets.')
+param keyVaultRoleAssignments roleAssignmentType[]?
+
+@description('Optional. The permissions that will be assigned to the Container Registry. The managed Identity will be assigned the permissions to get and list images.')
+param registryRoleAssignments roleAssignmentType[]?
+
+// network related parameters
+// -------------------------
+@description('Optional. Deploy resources in a virtual network and use it for private endpoints.')
+param deployInVnet bool = false
+
+@description('Conditional. A new private DNS Zone will be created. Setting to `false` requires an existing private DNS zone `privatelink.vaultcore.azure.net`. Required if `deployInVnet` is `true`.')
```

Comment:
This came up already discussing with @AlexanderSehr although we probably did not get to a final statement yet.
I'm curios to hear your thoughts a s well:
Should a Conditional parameter have a default value? We enforce a "Required if" statement for all Conditional parameter descriptions. Is it really "Required if" if we anyway have a default value for it?

Reply:
I'll hold my thoughts until @ReneHezser gave his thoughts 😄

Reply:
I understand the concerns. Technically, it is not required if it has a default value.
Since it is relevant only under circumstances, I pointed this out with "Required if", while still offering a default value.
Should we think about e.g. an "Applies if" in addition to a "Required if" to point out the scope of the parameter?

Reply:
I'd not want to overcomplicate this (by adding yet another case). @eriqua and I come from two different angles when it comes to this matter. @eriqua I'm not sure how we best continus from here. I guess we could discuss it in a call amongst the 3 of us? Or put it up for debate in the larger group? Whatever it is - getting to this 'final statement', you referred to before, would be great to get to and document (in the specs). Let me know you thoughts :)

Reply:
@AlexanderSehr, @ReneHezser works for me, probably a sync between the 3 of us would be enough. I don't want to block the PR on this, we have the same in many other places, so once discussed and documented we can, if ever, align the implementation as well. I just saw it and thought about mentioning. You can resolve this comment if you agree 🙂

Reply:
I put this topic on the agenda for today's call.